pkg/controller/nodes/task/taskexec_context.go

```diff
@@ -136,7 +136,15 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 		}
 	}
 
-	uniqueID, err := utils.FixedLengthUniqueIDForParts(IDMaxLength, nCtx.NodeExecutionMetadata().GetOwnerID().Name, currentNodeUniqueID, strconv.Itoa(int(id.RetryAttempt)))
+	length := IDMaxLength
+	if l := plugin.GetProperties().GeneratedNameMaxLength; l != nil {
+		length = *l
+		if length <= 0 {
+			return nil, errors.Errorf(errors.RuntimeExecutionError, nCtx.NodeID(), "GeneratedNameMaxLength shoud greater then 0")
```

Comment:
I think this should fail earlier.. maybe at plug-in registration/initialization...